src/FormModel.php

```diff
@@ -345,7 +356,13 @@ private function readProperty(string $attribute)
 
     private function writeProperty(string $attribute, $value): void
     {
-        if ($this->isPublicAttribute($attribute)) {
+        $setter = 'set' . $this->inflector->camelize($attribute);
+        if (is_callable([$this, $setter])) {
+            $method = new ReflectionMethod(static::class, $setter);
+            if ($method->getNumberOfRequiredParameters() === 1) {
+                $this->$setter($value);
+            }
+        } else if ($this->isPublicAttribute($attribute)) {
```

Comment:
```suggestion
        }
        if ($this->isPublicAttribute($attribute)) {
```

Reply:
Ideally we need `else if`, but I see the problem. The first `if` can return true and then execution will be exited when checking for parameters. So I need also to return if nested `if` did the work.

Reply:
Why the second `return` is needed when count parameters is not equals to 1?
Maybe just avoid to checking it?

Reply:
If I replace `else if` with `if` than if setter is available and correct (has required parameter) it will be called, after which also property will be written directly as we don't terminate execution. On the other if don't check for parameters we can have `setParameter()` method which has the correct naming, but is not intended to be used for setting or some other thing internally in the application, and consequently it doesn't have 1 required parameter to be used as setter and will have runtime exception. With the check we will just ignore it and fallback to setting attribute directly.

Reply:
I mean something like:
```php
if (is_callable([$this, $setter])) {
    $method = new ReflectionMethod(static::class, $setter);
    if ($method->getNumberOfRequiredParameters() === 1) {
        $this->$setter($value);
        return;
    }
}

if ($this->isPublicAttribute($attribute)) {
    $this->$attribute = $value;
} else {
    $setter = fn($class, $attribute, $value) => $class->$attribute = $value;
    $setter = Closure::bind($setter, null, $this);

    $setter($this, $attribute, $value);
}
```

Reply:
When `elseif` will remain the property will intact if form will have not correct setter.

I think this mechanism should work by this rules:
1. Use setter if available
2. Use directly property